folio-spring-base/src/test/java/org/folio/spring/client/EnrichUrlAndHeadersClientTest.java

```diff
@@ -0,0 +1,188 @@
+      is(expected)
+    );
+  }
+
+  @Test
+  void testHeaderPreparationWithNoLanguage() {
+    Request request = mock(Request.class);
+    when(request.headers())
+      .thenReturn(
+        Map.of(
+          "a", List.of("a-val"),
+          "b", List.of("b-val"),
+          "c", List.of("c-val")
+        )
+      );
+
+    FolioExecutionContext context = mock(FolioExecutionContext.class);
+    when(context.getOkapiHeaders()).thenReturn(Map.of("z", List.of("z-val")));
+    when(context.getAllHeaders())
+      .thenReturn(
+        Map.of("misc-1", List.of("misc-1-val"), "misc-2", List.of("misc-2-val"))
+      );
+
+    assertThat(
+      EnrichUrlAndHeadersClient.prepareHeaders(request, context),
+      is(
+        allOf(
+          hasEntry("a", List.of("a-val")),
+          hasEntry("b", List.of("b-val")),
+          hasEntry("c", List.of("c-val")),
+          hasEntry("z", List.of("z-val")),
+          is(aMapWithSize(4))
+        )
+      )
+    );
+  }
+
+  @Test
+  void testHeaderPreparationWithLanguage() {
+    Request request = mock(Request.class);
+    when(request.headers())
+      .thenReturn(
+        Map.of(
+          "a", List.of("a-val"),
+          "b", List.of("b-val"),
+          "c", List.of("c-val")
+        )
+      );
+
+    FolioExecutionContext context = mock(FolioExecutionContext.class);
+    when(context.getOkapiHeaders()).thenReturn(Map.of("z", List.of("z-val")));
+    when(context.getAllHeaders())
+      .thenReturn(
+        Map.of(
+          "misc-1", List.of("misc-1-val"),
+          "accept-language", List.of("en-US,en;q=0.9"),
+          "misc-2", List.of("misc-2-val")
+        )
+      );
+
+    assertThat(
+      EnrichUrlAndHeadersClient.prepareHeaders(request, context),
+      is(
+        allOf(
+          hasEntry("a", List.of("a-val")),
+          hasEntry("b", List.of("b-val")),
+          hasEntry("c", List.of("c-val")),
+          hasEntry("z", List.of("z-val")),
+          hasEntry("Accept-Language", List.of("en-US,en;q=0.9")),
+          is(aMapWithSize(5))
+        )
+      )
+    );
+  }
+
+  @Test
+  void testRequestExecution() throws IOException {
+    final Request request = Request.create(
+      Request.HttpMethod.GET,
+      "http://test-url",
+      Map.of("a", List.of("a-val")),
+      Request.Body.create("test-data"),
+      null
+    );
+
+    final FolioExecutionContext context = mock(FolioExecutionContext.class);
+    when(context.getOkapiUrl()).thenReturn("http://okapi");
+    when(context.getOkapiHeaders()).thenReturn(Map.of("z", List.of("z-val")));
+    when(context.getAllHeaders())
+      .thenReturn(
+        Map.of(
+          "misc-1", List.of("misc-1-val"),
+          "accept-language", List.of("en-US,en;q=0.9"),
+          "misc-2", List.of("misc-2-val")
+        )
+      );
+
+    final Request.Options options = new Request.Options();
+
+    final EnrichUrlAndHeadersClient client = new EnrichUrlAndHeadersClient(
+      context,
+      null
+    );
```

Comment:
Please avoid excessive line-wrapping. It doesn't necessarily enhance readability and can, in certain cases, reduce it.

Reply:
Please update your PR with avoiding line-wrapping where it possible.